`panic: runtime error: invalid memory address or nil pointer` when running `apply` step

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.2.8

### AzureRM Provider Version

3.21.1

### Affected Resource(s)/Data Source(s)

azurerm_key_vault.key_vault, azurerm_windows_virtual_machine_scale_set, azurerm_monitor_diagnostic_setting

### Terraform Configuration Files

```hcl
https://www.dropbox.com/sh/awtojtlsvvq8739/AACMR4S62j3yDE8TgntV3yMOa?dl=0
```


### Debug Output/Panic Output

```shell
2022-09-07T13:32:22.5711013Z ##[section]Starting: Terraform apply
2022-09-07T13:32:22.5726945Z ==============================================================================
2022-09-07T13:32:22.5727271Z Task         : Terraform
2022-09-07T13:32:22.5727677Z Description  : Execute terraform commands to manage resources on AzureRM, Amazon Web Services(AWS) and Google Cloud Platform(GCP)
2022-09-07T13:32:22.5728113Z Version      : 2.203.0
2022-09-07T13:32:22.5728340Z Author       : Microsoft Corporation
2022-09-07T13:32:22.5728672Z Help         : [Learn more about this task](https://aka.ms/AAf0uqr)
2022-09-07T13:32:22.5729042Z ==============================================================================
2022-09-07T13:32:22.7437167Z [command]/opt/hostedtoolcache/terraform/1.2.8/x64/terraform providers
2022-09-07T13:32:23.6438027Z 
2022-09-07T13:32:23.6439039Z Providers required by configuration:
2022-09-07T13:32:23.6439552Z .
2022-09-07T13:32:23.6440685Z ├── provider[registry.terraform.io/hashicorp/azurerm] <= 4.0.0
2022-09-07T13:32:23.6441600Z └── provider[registry.terraform.io/hashicorp/random] <= 4.0.0
2022-09-07T13:32:23.6442638Z 
2022-09-07T13:32:23.6443079Z Providers required by state:
2022-09-07T13:32:23.6443326Z 
2022-09-07T13:32:23.6443712Z     provider[registry.terraform.io/hashicorp/azurerm]
2022-09-07T13:32:23.6444012Z 
2022-09-07T13:32:23.6444393Z     provider[registry.terraform.io/hashicorp/random]
2022-09-07T13:32:23.6444683Z 
2022-09-07T13:32:23.6450785Z [command]/opt/hostedtoolcache/terraform/1.2.8/x64/terraform validate
2022-09-07T13:32:25.0306209Z [33m╷[0m[0m
2022-09-07T13:32:25.0307188Z [33m│[0m [0m[1m[33mWarning: [0m[0m[1mDeprecated attribute[0m
2022-09-07T13:32:25.0307789Z [33m│[0m [0m
2022-09-07T13:32:25.0309266Z [33m│[0m [0m[0m  on logs_and_diagnostics.tf line 20, in resource "azurerm_monitor_diagnostic_setting" "diagnostic_setting_logic_app":
2022-09-07T13:32:25.0310278Z [33m│[0m [0m  20:     for_each = data.azurerm_monitor_diagnostic_categories.logic_app_wf[4m.logs[0m[0m
2022-09-07T13:32:25.0310823Z [33m│[0m [0m
2022-09-07T13:32:25.0311392Z [33m│[0m [0mThe attribute "logs" is deprecated. Refer to the provider documentation for
2022-09-07T13:32:25.0311879Z [33m│[0m [0mdetails.
2022-09-07T13:32:25.0312222Z [33m│[0m [0m
2022-09-07T13:32:25.0312667Z [33m│[0m [0m(and 5 more similar warnings elsewhere)
2022-09-07T13:32:25.0313060Z [33m╵[0m[0m
2022-09-07T13:32:25.0313539Z [32m[1mSuccess![0m The configuration is valid, but there were some
2022-09-07T13:32:25.0313892Z validation warnings as shown above.
2022-09-07T13:32:25.0314248Z [0m
2022-09-07T13:32:25.0386357Z [command]/opt/hostedtoolcache/terraform/1.2.8/x64/terraform apply -auto-approve -lock-timeout=10m dev.plan
2022-09-07T13:32:25.6530667Z [33m╷[0m[0m
2022-09-07T13:32:25.6531699Z [33m│[0m [0m[1m[33mWarning: [0m[0m[1m"use_microsoft_graph": [DEPRECATED] This field now defaults to `true` and will be removed in v1.3 of Terraform Core due to the deprecation of ADAL by Microsoft.[0m
2022-09-07T13:32:25.6532501Z [33m│[0m [0m
2022-09-07T13:32:25.6532971Z [33m│[0m [0m[0m
2022-09-07T13:32:25.6533372Z [33m╵[0m[0m
2022-09-07T13:32:31.4859677Z [0m[1mazurerm_key_vault.key_vault: Modifying... [id=/subscriptions/4c307ec4-475a-4ec0-84f6-bf5330af9e31/resourceGroups/s-weu-dev-cq-shd-rg/providers/Microsoft.KeyVault/vaults/s-weu-dev-cq-kv][0m[0m
2022-09-07T13:32:31.5063267Z [0m[1mazurerm_monitor_diagnostic_setting.diagnostic_setting_data_factory: Modifying... [id=/subscriptions/4c307ec4-475a-4ec0-84f6-bf5330af9e31/resourceGroups/s-weu-dev-cq-rg/providers/Microsoft.DataFactory/factories/s-weu-dev-cq-df|s-weu-dev-cq-ds][0m[0m
2022-09-07T13:32:31.5197045Z [0m[1mazurerm_windows_virtual_machine_scale_set.vmss: Modifying... [id=/subscriptions/4c307ec4-475a-4ec0-84f6-bf5330af9e31/resourceGroups/s-weu-dev-cq-shd-rg/providers/Microsoft.Compute/virtualMachineScaleSets/dev-vmss][0m[0m
2022-09-07T13:32:31.5589049Z [0m[1mazurerm_monitor_diagnostic_setting.diagnostic_setting_dbw: Modifying... [id=/subscriptions/4c307ec4-475a-4ec0-84f6-bf5330af9e31/resourceGroups/s-weu-dev-cq-rg/providers/Microsoft.Databricks/workspaces/s-weu-dev-cq-dbw|s-weu-dev-cq-ds][0m[0m
2022-09-07T13:32:31.6390015Z [33m╷[0m[0m
2022-09-07T13:32:31.6390874Z [33m│[0m [0m[1m[33mWarning: [0m[0m[1mDeprecated attribute[0m
2022-09-07T13:32:31.6391449Z [33m│[0m [0m
2022-09-07T13:32:31.6415861Z [33m│[0m [0m[0m  on logs_and_diagnostics.tf line 49, in resource "azurerm_monitor_diagnostic_setting" "diagnostic_setting_data_factory":
2022-09-07T13:32:31.6416992Z [33m│[0m [0m  49:     for_each = data.azurerm_monitor_diagnostic_categories.data_factory[4m.logs[0m[0m
2022-09-07T13:32:31.6417662Z [33m│[0m [0m
2022-09-07T13:32:31.6418343Z [33m│[0m [0mThe attribute "logs" is deprecated. Refer to the provider documentation for
2022-09-07T13:32:31.6419030Z [33m│[0m [0mdetails.
2022-09-07T13:32:31.6419522Z [33m│[0m [0m
2022-09-07T13:32:31.6420110Z [33m│[0m [0m(and 3 more similar warnings elsewhere)
2022-09-07T13:32:31.6420653Z [33m╵[0m[0m
2022-09-07T13:32:31.6421206Z [31m╷[0m[0m
2022-09-07T13:32:31.6422321Z [31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
2022-09-07T13:32:31.6422804Z [31m│[0m [0m
2022-09-07T13:32:31.6423309Z [31m│[0m [0m[0m  with azurerm_windows_virtual_machine_scale_set.vmss,
2022-09-07T13:32:31.6423967Z [31m│[0m [0m  on compute.tf line 120, in resource "azurerm_windows_virtual_machine_scale_set" "vmss":
2022-09-07T13:32:31.6424649Z [31m│[0m [0m 120: resource "azurerm_windows_virtual_machine_scale_set" "vmss" [4m{[0m[0m
2022-09-07T13:32:31.6425098Z [31m│[0m [0m
2022-09-07T13:32:31.6425642Z [31m│[0m [0mThe plugin encountered an error, and failed to respond to the
2022-09-07T13:32:31.6426248Z [31m│[0m [0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2022-09-07T13:32:31.6426826Z [31m│[0m [0mcontain more details.
2022-09-07T13:32:31.6427194Z [31m╵[0m[0m
2022-09-07T13:32:31.6427539Z [31m╷[0m[0m
2022-09-07T13:32:31.6427989Z [31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
2022-09-07T13:32:31.6428436Z [31m│[0m [0m
2022-09-07T13:32:31.6428974Z [31m│[0m [0m[0m  with azurerm_monitor_diagnostic_setting.diagnostic_setting_data_factory,
2022-09-07T13:32:31.6429737Z [31m│[0m [0m  on logs_and_diagnostics.tf line 43, in resource "azurerm_monitor_diagnostic_setting" "diagnostic_setting_data_factory":
2022-09-07T13:32:31.6430518Z [31m│[0m [0m  43: resource "azurerm_monitor_diagnostic_setting" "diagnostic_setting_data_factory" [4m{[0m[0m
2022-09-07T13:32:31.6431002Z [31m│[0m [0m
2022-09-07T13:32:31.6431492Z [31m│[0m [0mThe plugin encountered an error, and failed to respond to the
2022-09-07T13:32:31.6432087Z [31m│[0m [0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2022-09-07T13:32:31.6432946Z [31m│[0m [0mcontain more details.
2022-09-07T13:32:31.6433311Z [31m╵[0m[0m
2022-09-07T13:32:31.6433649Z [31m╷[0m[0m
2022-09-07T13:32:31.6434097Z [31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
2022-09-07T13:32:31.6434539Z [31m│[0m [0m
2022-09-07T13:32:31.6435040Z [31m│[0m [0m[0m  with azurerm_monitor_diagnostic_setting.diagnostic_setting_dbw,
2022-09-07T13:32:31.6435784Z [31m│[0m [0m  on logs_and_diagnostics.tf line 101, in resource "azurerm_monitor_diagnostic_setting" "diagnostic_setting_dbw":
2022-09-07T13:32:31.6436526Z [31m│[0m [0m 101: resource "azurerm_monitor_diagnostic_setting" "diagnostic_setting_dbw" [4m{[0m[0m
2022-09-07T13:32:31.6436994Z [31m│[0m [0m
2022-09-07T13:32:31.6437682Z [31m│[0m [0mThe plugin encountered an error, and failed to respond to the
2022-09-07T13:32:31.6438294Z [31m│[0m [0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2022-09-07T13:32:31.6438826Z [31m│[0m [0mcontain more details.
2022-09-07T13:32:31.6439189Z [31m╵[0m[0m
2022-09-07T13:32:31.6447424Z [31m╷[0m[0m
2022-09-07T13:32:31.6447924Z [31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
2022-09-07T13:32:31.6448363Z [31m│[0m [0m
2022-09-07T13:32:31.6449057Z [31m│[0m [0m[0m  with azurerm_key_vault.key_vault,
2022-09-07T13:32:31.6449631Z [31m│[0m [0m  on main.tf line 81, in resource "azurerm_key_vault" "key_vault":
2022-09-07T13:32:31.6450207Z [31m│[0m [0m  81: resource "azurerm_key_vault" "key_vault" [4m{[0m[0m
2022-09-07T13:32:31.6450634Z [31m│[0m [0m
2022-09-07T13:32:31.6451119Z [31m│[0m [0mThe plugin encountered an error, and failed to respond to the
2022-09-07T13:32:31.6451716Z [31m│[0m [0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2022-09-07T13:32:31.6452225Z [31m│[0m [0mcontain more details.
2022-09-07T13:32:31.6452588Z [31m╵[0m[0m
2022-09-07T13:32:31.6642750Z 
2022-09-07T13:32:31.6644129Z Stack trace from the terraform-provider-azurerm_v3.21.1_x5 plugin:
2022-09-07T13:32:31.6644570Z 
2022-09-07T13:32:31.6645095Z panic: runtime error: invalid memory address or nil pointer dereference
2022-09-07T13:32:31.6645713Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x47e6605]
2022-09-07T13:32:31.6646146Z 
2022-09-07T13:32:31.6646554Z goroutine 107 [running]:
2022-09-07T13:32:31.6647541Z github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.resourceWindowsVirtualMachineScaleSetUpdate(0xc002622680, {0x57ef6e0?, 0xc0001fe380})
2022-09-07T13:32:31.6648702Z 	github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/windows_virtual_machine_scale_set_resource.go:526 +0x7c5
2022-09-07T13:32:31.6652220Z github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x6925580?, {0x6925580?, 0xc002747c80?}, 0xd?, {0x57ef6e0?, 0xc0001fe380?})
2022-09-07T13:32:31.6653024Z 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:729 +0x178
2022-09-07T13:32:31.6653936Z github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0008c4b60, {0x6925580, 0xc002747c80}, 0xc00221a0d0, 0xc002622400, {0x57ef6e0, 0xc0001fe380})
2022-09-07T13:32:31.6654768Z 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:847 +0x82c
2022-09-07T13:32:31.6655639Z github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0017bc198, {0x6925580?, 0xc002747bc0?}, 0xc0026ec640)
2022-09-07T13:32:31.6656453Z 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:1021 +0xe3c
2022-09-07T13:32:31.6657313Z github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0016259a0, {0x6925580?, 0xc002747650?}, 0xc000f016c0)
2022-09-07T13:32:31.6658081Z 	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:813 +0x4fc
2022-09-07T13:32:31.6659007Z github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5dcf7e0?, 0xc0016259a0}, {0x6925580, 0xc002747650}, 0xc002756120, 0x0)
2022-09-07T13:32:31.6660238Z 	github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
2022-09-07T13:32:31.6660859Z google.golang.org/grpc.(*Server).processUnaryRPC(0xc000246c40, {0x6936ca0, 0xc000e02b60}, 0xc0027545a0, 0xc0017ddf80, 0xa99eb80, 0x0)
2022-09-07T13:32:31.6661477Z 	google.golang.org/grpc@v1.47.0/server.go:1283 +0xcfd
2022-09-07T13:32:31.6661938Z google.golang.org/grpc.(*Server).handleStream(0xc000246c40, {0x6936ca0, 0xc000e02b60}, 0xc0027545a0, 0x0)
2022-09-07T13:32:31.6662411Z 	google.golang.org/grpc@v1.47.0/server.go:1620 +0xa1b
2022-09-07T13:32:31.6662782Z google.golang.org/grpc.(*Server).serveStreams.func1.2()
2022-09-07T13:32:31.6663163Z 	google.golang.org/grpc@v1.47.0/server.go:922 +0x98
2022-09-07T13:32:31.6663554Z created by google.golang.org/grpc.(*Server).serveStreams.func1
2022-09-07T13:32:31.6663933Z 	google.golang.org/grpc@v1.47.0/server.go:920 +0x28a
2022-09-07T13:32:31.6664110Z 
2022-09-07T13:32:31.6664582Z Error: The terraform-provider-azurerm_v3.21.1_x5 plugin crashed!
2022-09-07T13:32:31.6664777Z 
2022-09-07T13:32:31.6665090Z This is always indicative of a bug within the plugin. It would be immensely
2022-09-07T13:32:31.6665899Z helpful if you could report the crash with the plugin's maintainers so that it
2022-09-07T13:32:31.6666321Z can be fixed. The output above should help diagnose the issue.
2022-09-07T13:32:31.6666500Z 
2022-09-07T13:32:31.6718556Z ##[error]Error: The process '/opt/hostedtoolcache/terraform/1.2.8/x64/terraform' failed with exit code 1
2022-09-07T13:32:31.6754308Z ##[section]Finishing: Terraform apply
```


### Expected Behaviour

Apply should have completed.

Both when testing on different stage (qa and prod) instead of dev, it works. It also works when testing in a different subscription.

### Actual Behaviour

Error: The terraform-provider-azurerm_v3.21.1_x5 plugin crashed!

### Steps to Reproduce

run `terraform apply`

It may or may not be related to the specific state-file being used, which is also supplied in the provided link.

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/compute/windows_virtual_machine_scale_set_resource.go
+++ b/internal/services/compute/windows_virtual_machine_scale_set_resource.go
@@ -531,8 +531,9 @@ func resourceWindowsVirtualMachineScaleSetUpdate(d *pluginsdk.ResourceData, meta
 			upgradePolicy.AutomaticOSUpgradePolicy = ExpandVirtualMachineScaleSetAutomaticUpgradePolicy(automaticRaw)
 
 			// however if this block has been changed then we need to pull it
-			// we can guarantee this always has a value since it'll have been expanded and thus is safe to de-ref
-			automaticOSUpgradeIsEnabled = *upgradePolicy.AutomaticOSUpgradePolicy.EnableAutomaticOSUpgrade
+			if upgradePolicy.AutomaticOSUpgradePolicy != nil && upgradePolicy.AutomaticOSUpgradePolicy.EnableAutomaticOSUpgrade != nil {
+				automaticOSUpgradeIsEnabled = *upgradePolicy.AutomaticOSUpgradePolicy.EnableAutomaticOSUpgrade
+			}
 		}
 
 		if d.HasChange("rolling_upgrade_policy") {

